server/core/dcb.cc

```diff
@@ -508,6 +512,13 @@ dcb_connect(SERVER *server, MXS_SESSION *session, const char *protocol)
         dcb_free_all_memory(dcb);
         return NULL;
     }
+    
+    /* Register upstream throttling callbacks */
+    if (dcb->high_water && dcb->low_water && dcb->high_water > dcb->low_water)
```

Comment:
There should be a check somewhere that ensures that high_water > low_water and if it isn't logs an error and sets both to 0. There should also be `MIN_WRITEQ_HIGH_WATER` and `MIN_WRITEQ_LOW_WATER` below which the values are not allowed to be set.